src/app/clusters/group-key-mgmt-server/group-key-mgmt-server.cpp

```diff
@@ -0,0 +1,410 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/CommandHandler.h>
+#include <app/common/gen/att-storage.h>
+#include <app/common/gen/attribute-id.h>
+#include <app/common/gen/attribute-type.h>
+#include <app/common/gen/attributes/Accessors.h>
+#include <app/common/gen/cluster-id.h>
+#include <app/common/gen/command-id.h>
+#include <app/util/af.h>
+#include <map>
+#include <set>
+#include <string.h>
+#include <tuple>
+#include <utility>
+
+constexpr size_t GROUP_KEY_ROOT_SIZE  = 16;
+constexpr size_t GROUP_KEY_EPOCH_SIZE = 8;
+constexpr uint8_t GROUP_KEY_SET_SIZE  = 3;
+
+using namespace chip;
+
+struct RootKey
+{
+    bool active                             = false;
+    uint8_t value[GROUP_KEY_ROOT_SIZE]      = { 0 };
+    uint8_t startTime[GROUP_KEY_EPOCH_SIZE] = { 0 };
+    uint8_t policy                          = 0;
+};
+
+struct KeySet
+{
+    uint8_t current = 0;
+    RootKey epoch[GROUP_KEY_SET_SIZE];
+};
+
+static std::map<std::tuple<uint16_t, uint16_t>, KeySet *> sGroupKeys;
+static std::set<std::tuple<uint16_t, GroupId, uint16_t>> sGroupStates;
```

Comment:
std::map/std::set not available on all embedded targets. This is an STL reliance. Consider employing simpler data structures to avoid reliance on std::map/std::set.